static/js/stream_popover.js

```diff
@@ -406,12 +406,24 @@ function build_move_topic_to_stream_popover(e, current_stream_id, topic_name) {
 
     hide_topic_popover();
 
-    function move_topic() {
-        const params = Object.fromEntries(
+    function get_params_from_form() {
+        return Object.fromEntries(
             $("#move_topic_form")
                 .serializeArray()
                 .map(({name, value}) => [name, value]),
         );
+    }
+
+    function update_submit_button_disabled_state(select_stream_id) {
+        const {current_stream_id, new_topic_name, old_topic_name} = get_params_from_form();
+
+        $("#move_topic_modal .dialog_submit_button")[0].disabled =
+            Number.parseInt(current_stream_id, 10) === Number.parseInt(select_stream_id, 10) &&
+            new_topic_name.trim() === old_topic_name.trim();
```

Comment:
It appears you've switched from a case-insensitive comparison to a case-sensitive comparison here; what's the thinking behind that? Topics are case-insensitive in Zulip (from an equality standpoint), but I suppose it needs to be possible to change the case when editing?

So perhaps this change is an improvement, but it should ideally be in a second commit with its own commit message.

Reply:
Got it! The rationale was twofold: 

1) The comparison that sets new_topic_name to undefined in the callback to message_edit.with_first_message_id (so that the server knows that no change has occurred) only performs a trim(), but the new_topic_name variable is not necessarily lowercase, making the behavior appear inconsistent, or at least unintuitive (ie. the condition for showing the error banner was case insensitive, but the comparison that decides whether to set new_topic_name to undefined appeared to be case sensitive)

2) In choosing between the two, it seemed that it would be confusing to the user to encounter a disabled submit button whenever they're just trying to change the case of the topic.

I suppose I'll put the change in a separate commit.

Reply:
Great! This makes sense. The separate commit is great, and just to highlight how splitting the commit was helpful: having it split out as its own change helped me realize that we probably want to document this decision in a code comment. 

Generally the philosophy is that one wants to document decisions like this one somewhere, and the location for that documentation varies with the situation. If it's a question one might have reading the code, it belongs in a code comment. If it's a question one would really only have about the change (E.g. the final state reads plainly as correct, but one might worry the change introduces a bug), those are best documented in a commit message. And things that are more at a higher level, like "Here's why I think we should do this rather than X totally different approach that might solve the problem" or "I double-checked that we didn't have other instances of this bug", are great as PR comments.

Reply:
Oh, thanks! That additional comment made a lot of sense to include. And this is a helpful heuristic for deciding where to put things; I'll definitely keep it in mind.